logback/src/main/java/com/linecorp/armeria/common/logback/RequestContextExportingAppender.java

```diff
@@ -229,7 +226,11 @@ public void stop() {
 
     @Override
     public void addAppender(Appender<ILoggingEvent> newAppender) {
-        aai.addAppender(newAppender);
+        if (newAppender instanceof AbstractSocketAppender) {
```

Comment:
`AbstractSocketAppender` approach looks like a general solution and can isolate the customizing logic to it. But it wouldn't be efficient since both `UnionMap` and `HashMap` are created for `AbstractSocketAppender`. How about adding a flag for whether to use `HashMap` or not? For example:
```java
private boolean needsHashMap;

public void addAppender(Appender<ILoggingEvent> newAppender) {
    if (newAppender instanceof AbstractSocketAppender) {
       needsHashMap = true;
    }
    ...
}

protected void append(ILoggingEvent eventObject) {
    if (needsHashMap) {
        mdcMap = new HashMap<>(contextMap);
        // mdcMap values can be overridden by originalMdcMap
        mdcMap.putAll(originalMdcMap);
    } else {
        mdcMap = new UnionMap<>(contextMap, originalMdcMap);
    }
```

Reply:
Yeah, I agree. There's no point to create `UnionMap` and then create `HashMap` of it. I've fixed this.